src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterEnum } from './types/filter';
+import { TodoModal } from './components/TodoModal';
+import { getTodos } from './api';
+
+const getFilteredTodos = (todos: Todo[], filter: FilterEnum, query = ''):Todo[] => {
+  let filterTodos = [...todos];
+
+  filterTodos = filterTodos.filter(todo => {
```

Comment:
The `filter` method returns a copy of the array, so you don't need to do it manually.